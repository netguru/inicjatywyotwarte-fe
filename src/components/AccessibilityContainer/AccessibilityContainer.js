import React from 'react'
import styled from 'styled-components'
import { ReactComponent as ContrastIcon } from 'assets/contrast_icon.svg'

const StyledToggleContrastButton = styled.a`
  height: 2rem;
  width: 5rem;
  color: ${props => props.theme.fontColorDark};
  font-size: 0.85rem;
  padding: 0 1rem;
  cursor: pointer;
  height: 2.5rem;
  transition: ${props => props.theme.buttonTransition};
  text-decoration: none;
  display: flex;
  align-items: center;
  margin-right: 0.3rem;

  svg {
    height: 60%;
    width: 60%;
  }
`
StyledToggleContrastButton.displayName = 'StyledToggleContrastButton';

function AccessibilityContainer ({theme, toggleContrast}) {
  return (
    <>
      <StyledToggleContrastButton onClick={toggleContrast}><ContrastIcon/>&nbsp;Kontrast</StyledToggleContrastButton>
    </>
  )
}

export default AccessibilityContainer;
