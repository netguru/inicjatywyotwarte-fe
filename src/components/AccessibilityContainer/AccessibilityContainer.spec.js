import React from 'react'
import AccessibilityContainer from './AccessibilityContainer'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  theme: jest.fn(),
  toggleContrast: jest.fn()
};

describe('AccessibilityContainer', () => {
  test('has button to toggle contrast', () => {
    const wrapper = shallow(<AccessibilityContainer {...baseProps} />);
    expect(wrapper.find('StyledToggleContrastButton').length).toEqual(1)
    expect(wrapper.find('StyledToggleContrastButton').text()).toEqual("\u00a0Kontrast")
  });

  test('calls function when button clicked', () => {
    const wrapper = shallow(<AccessibilityContainer {...baseProps} />);
    wrapper.find('StyledToggleContrastButton').simulate('click');
    expect(baseProps.toggleContrast).toHaveBeenCalledTimes(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<AccessibilityContainer {...baseProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


