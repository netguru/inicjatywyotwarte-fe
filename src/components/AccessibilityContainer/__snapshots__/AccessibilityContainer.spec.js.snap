// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessibilityContainer match snapshot 1`] = `
<Fragment>
  <StyledToggleContrastButton
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <ForwardRef(SvgContrastIcon) />
    Â Kontrast
  </StyledToggleContrastButton>
</Fragment>
`;
